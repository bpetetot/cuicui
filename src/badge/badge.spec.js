/* eslint-disable react/jsx-filename-extension */
/* eslint-env jest */
import snap from 'snap'

import Badge from './badge'

const snapshot = props => snap(Badge)({ ...props })

describe('badge', () => {
  it('should add children', snapshot({ children: 'hello world' }))
  it('should add custom className', snapshot({ children: 'hello world', className: 'custom' }))
  it('should add custom style', snapshot({ children: 'hello world', style: { color: 'red' } }))
  it('should display like a pill', snapshot({ children: 'hello world', pill: true }))
  it('should take react callback', snapshot({ children: 'hello world', onClick: () => ({}) }))
})
