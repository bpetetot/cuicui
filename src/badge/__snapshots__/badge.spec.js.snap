// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`badge should add children 1`] = `
<span
  className="cc-badge cc-badge-default"
  style={Object {}}
>
  hello world
</span>
`;

exports[`badge should add custom className 1`] = `
<span
  className="cc-badge cc-badge-default custom"
  style={Object {}}
>
  hello world
</span>
`;

exports[`badge should add custom style 1`] = `
<span
  className="cc-badge cc-badge-default"
  style={
    Object {
      "color": "red",
    }
  }
>
  hello world
</span>
`;

exports[`badge should display like a pill 1`] = `
<span
  className="cc-badge cc-badge-default cc-badge-pill"
  style={Object {}}
>
  hello world
</span>
`;

exports[`badge should take react callback 1`] = `
<span
  className="cc-badge cc-badge-default"
  onClick={[Function]}
  style={Object {}}
>
  hello world
</span>
`;
